great_expectations/datasource/data_connector/configured_asset_azure_data_connector.py

```diff
@@ -0,0 +1,109 @@
+    def __init__(
+        self,
+        name: str,
+        datasource_name: str,
+        container: str,
+        assets: dict,
+        execution_engine: Optional[ExecutionEngine] = None,
+        default_regex: Optional[dict] = None,
+        sorters: Optional[list] = None,
+        prefix: str = "",
+        delimiter: str = "/",
+        max_keys: int = 1000,
+        azure_options: Optional[dict] = None,
+        batch_spec_passthrough: Optional[dict] = None,
+    ):
+        logger.debug(f'Constructing ConfiguredAssetAzureDataConnector "{name}".')
+
+        super().__init__(
+            name=name,
+            datasource_name=datasource_name,
+            execution_engine=execution_engine,
+            assets=assets,
+            default_regex=default_regex,
+            sorters=sorters,
+            batch_spec_passthrough=batch_spec_passthrough,
+        )
+        self._container = container
+        self._prefix = os.path.join(prefix, "")
+        self._delimiter = delimiter
+        self._max_keys = max_keys
+
+        if azure_options is None:
+            azure_options = {}
+
+        try:
+            # TODO(cdkini): Implement various methods of instantiation and authentication
+            self._azure = BlobServiceClient(**azure_options)
+            # self._azure = BlobServiceClient.from_connection_string()
+        except (TypeError, AttributeError):
+            raise ImportError(
+                "Unable to load Azure BlobServiceClient (it is required for ConfiguredAssetAzureDataConnector)."
+            )
+
+    def build_batch_spec(self, batch_definition: BatchDefinition) -> AzureBatchSpec:
+        batch_spec: PathBatchSpec = super().build_batch_spec(
+            batch_definition=batch_definition
+        )
+        return AzureBatchSpec(batch_spec)
+
+    # FIXME(cdkini): Currently breaks DataConnect.self_check()
+    # Let's identify what actually goes in query_options
+    def _get_data_reference_list_for_asset(self, asset: Optional[Asset]) -> List[str]:
+        query_options: dict = {
+            "container": self._container,
+            "name_starts_with": self._prefix,
+            "delimiter": self._delimiter,
+        }
+        if asset is not None:
+            if asset.bucket:
+                query_options["container"] = asset.bucket
+            if asset.prefix:
+                query_options["name_starts_with"] = asset.prefix
+            if asset.delimiter:
+                query_options["delimiter"] = asset.delimiter
+            # if asset.max_keys:
+            #    query_options["MaxKeys"] = asset.max_keys
+
+        path_list: List[str] = [
+            key
+            for key in list_azure_keys(
+                azure=self._azure,
+                container=self._container,
+                query_options=query_options,
+                # iterator_dict={}, # TODO(cdkini): I don't think we need this? Open to check.
+                # recursive=False,
```

Comment:
I don't think `iterator_dict` is needed here but I also don't really understand how it is used in S3. `recursive` seems valid.

Ex:
```
name_starts_with="2018", recursive=True # Gives me everything in the 2018/ dir as well as anything further nested
names_starts_with="", recursive=False # Gives me just blobs in the root directory (avoids exploring 2018/ dir)
```